kura/org.eclipse.kura.ble.provider/src/main/java/org/eclipse/kura/internal/ble/beacon/BluetoothLeBeaconScannerImpl.java

```diff
@@ -15,15 +15,16 @@
 import org.eclipse.kura.bluetooth.le.beacon.BluetoothLeBeaconDecoder;
 import org.eclipse.kura.bluetooth.le.beacon.BluetoothLeBeaconScanner;
 import org.eclipse.kura.bluetooth.le.beacon.listener.BluetoothLeBeaconListener;
+import org.eclipse.kura.internal.ble.BluetoothLeAdapterImpl;
 
 public class BluetoothLeBeaconScannerImpl<T extends BluetoothLeBeacon> implements BluetoothLeBeaconScanner<T> {
 
-    private final BluetoothLeAdapter adapter;
+    private final BluetoothLeAdapterImpl adapter;
     private final BluetoothLeBeaconDecoder<T> decoder;
```

Comment:
Moving the `startDiscovery` in the API, we can leave the `BluetoothLeAdapter`. So, we'll pass the interface, not the implementation.